File: AssignCookies.py
class Solution:
    def findContentChildren(self, g, s) -> int:
        g.sort()
        s.sort()
        i, j = 0, 0
        while i < len(g) and j < len(s):
            if g[i] <= s[j]:
                i += 1
            j += 1
        return i


if __name__ == '__main__':
    g = [int(num) for num in input().split()]
    s = [int(num) for num in input().split()]
    solution = Solution()
    print(solution.findContentChildren(g, s))

File: AssignCookies.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        
        int i = 0, j = 0;
        while(i < g.size() && j < s.size()) {
        	if(s[j] < g[i]) j++;
        	else {
        		i++;
        		j++;
			}
		}
		return i;
    }
};

int main() {
	
	int n;
	cin>>n;
	
	vector<int> g;
	for(int i = 0; i < n; i++) {
		int num;
		cin>>num;
		g.push_back(num);
	} 
	
	cin>>n;
	vector<int> s;
	for(int i = 0; i < n; i++) {
		int num;
		cin>>num;
		s.push_back(num);
	}
	
	Solution *solution = new Solution();
	cout<<solution->findContentChildren(g, s);
	
	return 0;
	
}
